File: 20210827_list/list.rb
def dump(a)
  if a.instance_of?(Array)
    a.each_with_index do |n, k|
      print "[#{k}] -> "
      p n
    end
  end
end

list = [];
list.push(1)
list.push(2)
list.push(3)
list.push(4)
list.push(5)

dump list
p list.length

list.delete_at 0
dump list
p list.length

sub = list.slice(2, 2)
dump sub
p sub.length

File: 20210827_list/list.py


def dump(a):
    for k in range(len(a)):
        print(f"[{k}] -> ", end = "")
        print(f"{a[k]}")

list = [];
list.append(1)
list.append(2)
list.append(3)
list.append(4)
list.append(5)

dump(list)
print(len(list))

del list[0]

dump(list)
print(len(list))

sub = list[2:4]
dump(sub)
print(len(sub))
